File: src/main/java/org/fabiomsr/javaxml/_00_serialization/Book.java
package org.fabiomsr.javaxml._00_serialization;

import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.Element;
import org.simpleframework.xml.Root;

/**
 * Created by fabiomsr on 2/7/16.
 */
@Root
public class Book {
    @Element
    private String title;

    @Attribute
    private int pages;

    public Book() {}

    public Book(String title, int pages) {
        this.title = title;
        this.pages = pages;
    }

    public String getTitle() {
        return title;
    }

    public int getPages() {
        return pages;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Book book = (Book) o;

        if (pages != book.pages) return false;
        return title != null ? title.equals(book.title) : book.title == null;

    }

    @Override
    public int hashCode() {
        int result = title != null ? title.hashCode() : 0;
        result = 31 * result + pages;
        return result;
    }
}

File: src/main/kotlin/org/fabiomsr/kotlinxml/_00_serialization/Book.kt
package org.fabiomsr.kotlinxml._00_serialization

import org.simpleframework.xml.Attribute
import org.simpleframework.xml.Element
import org.simpleframework.xml.ElementList
import org.simpleframework.xml.Root

/**
 * Created by fabiomsr on 2/7/16.
 */
@Root
data class Book( @field:Element var title: String,
                 @field:Attribute var pages: Int) {

    constructor( ) : this("", 0) {}
}